xain_fl/coordinator/coordinator.py

```diff
@@ -34,61 +34,74 @@ class Coordinator:  # pylint: disable=too-many-instance-attributes
+
+        - ``ROUND``: A round is currently in progress. During a round
+          the important messages the coordinator can receive are
+          :class:`~.coordinator_pb2.StartTrainingRoundRequest` and
+          :class:`~.coordinator_pb2.EndTrainingRoundRequest`.  Since
+          participants are selected for rounds or not, they can be
+          advertised either ROUND or STANDBY accordingly.
+
+        - ``FINISHED``: The training session has ended and
+          participants should disconnect from the coordinator.
+
+    States are exchanged during heartbeats so that both coordinators
+    and participants can react to each others state change.
 
     The flow of the Coordinator:
+
         1. The coordinator is started and waits for enough participants to join. `STANDBY`.
         2. Once enough participants are connected the coordinator starts the rounds. `ROUND N`.
         3. Repeat step 2. for the given number of rounds
         4. The training session is over and the coordinator is ready to shutdown. `FINISHED`.
 
     Note:
+
         :class:`~.coordinator_pb2.RendezvousRequest` is always allowed
         regardless of which state the coordinator is on.
 
-    Constants:
-        DEFAULT_AGGREGATOR (:class:`~.Aggregator`): if no Aggregator instance is provided
-            during initialisation, then :class:`~.FederatedAveragingAgg` in used.
-        DEFAULT_CONTROLLER (:class:`~.Controller`): if no Controller instance is provided
-            during initialisation, then :class:`~.RandomController` in used.
-
-    Attributes:
-        num_rounds (:obj:`int`, optional): The number of rounds of the training
-            session. Defaults to 10.
-        minimum_participants_in_round (:obj:`int`, optional): The minimum number of
-            participants that participate in a round. Defaults to 1.
-        fraction_of_participants (:obj:`float`, optional): The fraction of total
-            connected participants to be selected in a single round. Defaults to 1.0,
-            meaning that all connected participants will be selected.
-            It must be in the (0.0, 1.0] interval.
-        weights (:obj:`list` of :class:`~numpy.ndarray`, optional): The weights of
-            the global model. Defaults to [].
-        epochs (:obj:`int`, optional): Number of training iterations local to
-            Participant.  Defaults to 0.
-        epochs_base (:obj:`int`, optional): Global number of epochs as of last
-            round. Defaults to 0.
-        aggregator: (:class:`~.Aggregator`, optional): The type of aggregation
-            to perform at the end of each round. Defaults to :class:`~.FederatedAveragingAgg`.
-        controller: (:class:`~.Controller`, optional): Controls how the Participants
-            are selected at the start of each round. Defaults to :class:`~.RandomController`.
-        """
+    Args:
+
+        num_rounds: The number of rounds of the training session
```

Comment:
why has the attributes typing been removed? is it going to be picked up automatically?

Reply:
Yes that's the point of this PR :)